### PR TITLE
Qt: Add "Grab Cursor" option

### PR DESCRIPTION
**Pull request by spycrab**

### CODE REVIEW

File: Source/Core/DolphinQt2/Config/Graphics/GeneralWidget.cpp [hunk 1]
```diff
@@ -264,6 +273,9 @@ void GeneralWidget::AddDescriptions()
   static const char* TR_HIDE_MOUSE_CURSOR_DESCRIPTION =
       QT_TR_NOOP("Hides the mouse cursor if it's on top of the emulation window.\n\nIf unsure, "
                  "leave this unchecked.");
+  static const char* TR_GRAB_MOUSE_CURSOR_DESCRIPTION =
+      QT_TR_NOOP("Grabs the mouse cursor if the emulation window is in focus.\n\nIf unsure, "
```
Comment (JosJuice) [inline Source/Core/DolphinQt2/Config/Graphics/GeneralWidget.cpp]: I don't think this is a clear enough explanation for everyone. We need something more explanatory than the word "grabs".

Comment (spycrab) [reply Source/Core/DolphinQt2/Config/Graphics/GeneralWidget.cpp]: Sure. Got any ideas?

Comment (JosJuice) [reply Source/Core/DolphinQt2/Config/Graphics/GeneralWidget.cpp]: Maybe something like "Prevents the mouse cursor from leaving the emulation windows when it is in focus."

Comment (spycrab) [reply Source/Core/DolphinQt2/Config/Graphics/GeneralWidget.cpp]: Well, it actually doesn't do that: It simply makes you unable to click on anything while the cursor is in focus. (Mostly useful when using an emulated Wiimote in windowed Mode)

Comment (JosJuice) [reply Source/Core/DolphinQt2/Config/Graphics/GeneralWidget.cpp]: Oh, okay. Well, I guess that goes to show why the description could be more detailed :)

Comment (riking) [reply Source/Core/DolphinQt2/Config/Graphics/GeneralWidget.cpp]: "Prevents issues with accidentally clicking outside the emulation window" ?

Comment (spycrab) [reply Source/Core/DolphinQt2/Config/Graphics/GeneralWidget.cpp]: Fine by me. Needs @MayImilae's approval though.

**Comment by MayImilae** [comment]
This doesn't seem to do much on macOS. With render to main window I can't click the buttons in the toolbar without clicking out and back in, but otherwise, this seems to just do nothing. :/

**Comment by spycrab** [comment]
@MayImilae Should work on macOS now.

**Comment by MayImilae** [comment]
So um, it kind of works on macOS, but not really. When I tried moving the mouse cursor off the window, it only caught it on the bottom and right sides. The top and left sides just let it leave the window.

Also after clicking out and clicking back in a few times during testing, the right side stopped blocking it as well, and only the bottom would stop it from leaving the window. No idea what that's about!

**Comment by MayImilae** [comment]
So um, for whatever reason, the buildbot is building this PR incorrectly. It thinks it is building 6620, and  shows it in the dmg, but when I extract it, it is pr6680, and the Grab Cursor setting from this PR is no where to be seen.



Until this is fixed, I can't test this on macOS anymore. :/

**Comment by MayImilae** [comment]
So after that rebase, it is now pr6673. ¯\\\_(ツ)_/¯

screen shot 2018-04-27 at 6 11 53 pm

**Comment by spycrab** [comment]
Gonna redo this at some point in time.

**Pull request closed without merging**